src/lib.rs

```diff
@@ -98,6 +96,49 @@ impl<Data: AsRef<[u8]>> Reader<Data> {
             .unwrap()
             .map(move |level| &self.input.as_ref()[level.offset as usize..(level.offset + level.length_bytes) as usize])
     }
+
+    pub fn supercompression_global_data(&self) -> &[u8] {
+        let header = self.header();
+        let start = header.sgd_byte_offset as usize;
+        let end = (header.sgd_byte_offset + header.sgd_byte_length) as usize;
+        self.input.as_ref()[start..end].try_into().unwrap()
+    }
+
+    pub fn data_format_descriptors(&self) -> impl Iterator<Item = BasicDataFormatDescriptor> {
+        let header = self.header();
+        let start = header.dfd_byte_offset as usize;
+        let length: u32 = u32::from_le_bytes(self.input.as_ref()[start..start + 4].try_into().unwrap());
+        assert_eq!(length, header.dfd_byte_length);
+        assert_eq!(length, header.kvd_byte_offset - header.dfd_byte_offset);
```

Comment:
Are these asserts defending against malformed input data or bugs in this library? If the former, we should try to check during `new` instead. If the latter, switch to `debug_assert`.

Reply:
I'll move the checks to new. Should be to check the validity of input data.

Reply:
Although, should we deny parsing a file at all if only part of it is corrupt?

Reply:
I think that's the correct default, and helps prevent the widespread dissemation of invalid data to begin with. It also makes the user's life easier, because they can rely on a `ktx2::Reader` always representing valid data, rather than having to validate every possible malformed component themselves and/or litter their code with `unwrap`s.